File: Week_02/144. 二叉树的前序遍历/preorderTraversal.cpp
class Solution {
public:
    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> ans;
        if(root == nullptr) {
            return ans;
        }
        std::stack<TreeNode*> treeStack;
        treeStack.push(root);
        while(!treeStack.empty()) {
            TreeNode* node = treeStack.top();
            treeStack.pop();
            if(node) {
                ans.push_back(node->val);
                if(node->right) {
                    treeStack.push(node->right);
                }
                if(node->left) {
                    treeStack.push(node->left);
                }
            }
        }
        return ans;
    }
};

File: Week_02/144. 二叉树的前序遍历/preorderTraversal.java
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        LinkedList<Integer> ans = new LinkedList<>();
        if(root == null) {
            return ans;
        }
        LinkedList<TreeNode> stack = new LinkedList<>();
        stack.add(root);
        while(!stack.isEmpty()) {
            TreeNode node = stack.pollLast();
            ans.add(node.val);
            if(node.right != null) {
                stack.add(node.right);
            }
            if(node.left != null) {
                stack.add(node.left);
            }
        }
        return ans;
    }
}